Explain the difference between user and system connection

skobyda opened the pull request:
Fixes https://github.com/cockpit-project/cockpit-machines/issues/792

## Clarify the difference between user and system connection

**System:**
- Ideal for server VMs
- VM will launch with root permissions
- Ideal networking support
- Permissions denied for disk images in home directories

**User session:**
- Good choice for desktop virtualization
- VM launched with unprivileged limited access, with the process and PTY owned by your user account
- Restrictions in networking (SLIRP-based emulation) and PCI device assignment
- Disk images can be stored in user home directory

![Screenshot from 2023-01-03 14-43-36](https://user-images.githubusercontent.com/42733240/210369317-fab6bfb1-098d-4bc4-aeed-adb9a9b1acaf.png)

KKoukiou reviewed (dismissed):
I like this but I am sure @garrett will have some comments. Let's wait for his feedback

garrett reviewed (dismissed):
This is a great start! Thanks for writing the text! My initial impression is that it's quite verbose. But this makes it easier to edit down, so it's wonderful for a first pass. :+1: 

Should we use [lists](https://www.patternfly.org/v4/components/list)? It would be quicker to scan and understand, and if we don't conform to full sentences, we could use fragments to get to the point more quickly. (That is: Lists should be consistent; either full sentences for all points on the screen at the same time, or just fragments. Both are correct, as long as it's consistent overall. [See PF's guidelines for more information](https://www.patternfly.org/v4/ux-writing/writing-design-guidelines-for-patternfly/#lists). The sentence/fragment rule is implied in PF's guidelines, but it's a general style everywhere, at least in English. There's a good reference in [Microsoft's style guide](https://learn.microsoft.com/en-us/style-guide/scannable-content/lists#punctuation), for example.)

What do you think of the following? I tried to distill it down to main points for each.

## System

- Ideal for server VMs
- VM will launch with root permissions
- Ideal networking support
- Permissions denied for disk images in home directories

## User session

- Good choice for desktop virtualization
- VM launched with unprivileged limited access, with the process and PTY owned by your user account
- Restrictions in networking (SLIRP-based emulation) and PCI device assignment
- Disk images are stored in user home directory

skobyda commented:
Updated the PR. The list and the suggested text looks good. 
Maybe just one little thing, we could say `Disk images can be stored in user home directory` instead of `Disk images are stored in user home directory`? Because The difference is that System VM cannot access disks inside users home, but user can still configure VM to access disk which is outside of home. Then there are also network disks and block disks, which might might also be outside of home directory.
Anyway, here is a new version:

![Screenshot from 2022-12-08 15-03-46](https://user-images.githubusercontent.com/42733240/206465997-c4131658-06d7-423e-bcf7-11a385b07abb.png)

garrett commented:
> Maybe just one little thing, we could say Disk images can be stored in user home directory instead of Disk images are stored in user home directory? Because The difference is that System VM cannot access disks inside users home, but user can still configure VM to access disk which is outside of home. Then there are also network disks and block disks, which might might also be outside of home directory.
Anyway, here is a new version:

Sure! Thanks for the suggestion.

Also, is there a way to reduce the left padding (or margin?) of the lists? It seems rather excessive... is it nested in any other component? If not, then perhaps [using a spacing utility for margin or padding](https://www.patternfly.org/v4/utilities/spacing) (depending on what the lists use) might be needed.

garrett commented:
The lists have both padding and margin. Reducing padding on the left to 0 seems to do the trick. And I also changed the width to 100%, as it seemed constricted on the right side too.

![image](https://user-images.githubusercontent.com/10246/206520721-4fd31849-6e5c-491c-8ea0-4d55ba5e650d.png)

The direction of the popover is also questionable, but I guess this is OK?

And it seems we have a regression for the alignment somehow...?!? Not sure if it's from this PR or in main like this.

![image](https://user-images.githubusercontent.com/10246/206521262-05e24cb6-f907-4b70-b4ae-e6f36a21218f.png)

skobyda commented:
> The lists have both padding and margin. Reducing padding on the left to 0 seems to do the trick. And I also changed the width to 100%, as it seemed constricted on the right side too.

Fixed
![Screenshot from 2023-01-02 13-39-28](https://user-images.githubusercontent.com/42733240/210235503-1772a3d4-fbf3-4216-a150-6051a99df53e.png)


> The direction of the popover is also questionable, but I guess this is OK?

The direction is chosen automatically by patternfly based on available space. If we hardcode it to place the popover to the right, it would not entirely fit in the page (see screenshot below). I think letting patternfly choose (in our case, it places popover to the bottom) is okay.
![Screenshot from 2023-01-02 13-39-55](https://user-images.githubusercontent.com/42733240/210235624-689b5a94-6557-48aa-b42c-e20b55148cae.png)


> And it seems we have a regression for the alignment somehow...?!? Not sure if it's from this PR or in main like this.

Fixed:
![Screenshot from 2023-01-02 13-39-11](https://user-images.githubusercontent.com/42733240/210235538-74fd66cf-198f-4d5c-8ccf-49556926e625.png)

martinpitt reviewed (approved):
LGTM, thank you! This may be release-notes material, if you'd like to see it mentioned, can you please update the description's screenshot to the current design and light mode?
